EIPS/eip-3416.md

```diff
@@ -0,0 +1,114 @@
+  * Let `GASUSED` be the gas used during the transaction execution/state transition.
+  * The `tx.origin` initially pays `GASPRICE * tx.gasLimit`, and gets refunded `GASPRICE * (tx.gasLimit - GASUSED)`
+  * The `block.coinbase` gains `GASPRICE * GASUSED`.
+
+The miners can still use a `greedy` strategy to include new transactions in the proposed blocks by adding the transactions ordered by larger `FEE_CAP` first. This is similar to how current blocks are filled, and is a consequence of `FEE_CAP` and `GAS_PREMIUM` being a positive linear function of each other.
+
+## Rationale 
+
+The rationale behind the premium being 50% of (fee cap - base fee) is that at any given point the average network sender has an average fee cap, and we assume that between base fee and fee cap the sender has no specific preference, as long as the transaction is included in some block. Then, the sender is happy with a median premium among this uniform range. Another justification can be that the user also knows that this new version of the pricing protocol for the complete block uses a median, then is fair for him to apply a median within his preferential range, assuming an uniform sampling there.
+
+## Backwards Compatibility
+
+The backward compatibility is very straightforward because there are no new fields added to the transactions. Pricing of the gas per block is still fast to compute but a little more complex. We split the EIP3416 upgrade into two phases with a transition period during which both legacy and EIP3416 transactions can be accepted so that compatibility with wallets and other ETH-adjacent software is maintained while their maintainers have time to upgrade to using the new transaction type. During this transition period legacy transactions are accepted and processed identically to the current implementation, with the only difference being that the amount of gas (gas limit) dedicated to processing legacy transactions is calculated as above and decreases over this period.
+
+## Test Cases
+
+## Security Considerations
+
+* Senders cannot manipulate the minimum fee because the minimum `BASE_FEE` is controlled by the miners with small increments or decrements on each new block proposed.
+* Above the `BASE_FEE` the senders have a very limited ability to manipulate and lower the final gas price they pay because they have to move the weighted median close to `BASE_FEE` and, as we know, this is a very robust statistic.
+* Miners have a very limited ability to manipulate and raise the final gas price paid by senders above `BASE_FEE` because to influence the final gas price they have to stuff fake transactions beyond the top 5% of the blocks. In average and currently, only the top 3% of the block is empty, so to fill-up 5% of the block they need to start dropping profitable transactions to reach 5%. Only beyond 5% of the top block gas they can start moving the median a little and the median is still a very robust statistic, not liable to being easily manipulated.
```

Comment:
Miners can drop the bottom 10% of lowest paying transactions and replace them with transactions that that pay an exorbitant fee to raise the median.  This has potential to fully offset the losses from those lowest 10% of transactions.

Reply:
The median is robust enough to handle a large percentage of Txs (<50%) replaced by miners Txs. And they have a penalty if they remove Txs beyond the average empty block utilization (currently around 3%). You mention below miners filling-up 50% of the block, that is a different scenario but might be not profitable.

Reply:
The miners cannot profit from exorbitant fake fees because the maximum user TXs are willing to pay is `fee_cap - base_fee` with `(fee_cap - base_fee)/2` used to compute the median premium. If the median surpases `fee_cap - base_fee` for a given user Tx then this Tx cannot be included in the block.